lib/afinter.c

```diff
@@ -496,3 +501,75 @@ afinter_global_deinit(void)
     }
   current_internal_source = NULL;
 }
+
+/****************************************************************************
+ * Live collection of internal log messages.
+ ****************************************************************************/
+
+gboolean
+afinter_start_live_collection(GString *result)
```

Comment:
The `result` variable is a coupling between `syslog-ng-ctl` functionality and `afinter`; and as such should not exists. Specially because there is a nice result information already conveyed as a return value, if the `success`/`failure` is not enough to represent the result, you could extend it via an enum describing the states, and not using strings those states.

Don't get me wrong, writing such errors is fine, but the string should be created in the correct layer (`syslog-ng-ctl` related functions).

Reply:
okay. I will add enum which will describe the states.